### PR TITLE
Fix building pkg/util/procfs on non-linux platforms

### LINKED ISSUE #393: We need a make release job
https://github.com/kubernetes/kubernetes/issues/30384 exposed the lack of a `make release` job.  We need one of those, post-submit if necessary.  @fejta @spxtr @ixdy

### PR DESCRIPTION
**Pull request by yujuhong**
This partially fixes https://github.com/kubernetes/test-infra/issues/393

/cc @ixdy @krousey @dims

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30723)

### CODE REVIEW

**Comment by brendandburns** [comment]
LGTM.  e2e tests are a flake due to out of CPU quota.  Will wait for that to get resolved before I ping the bot again.

**Comment by yujuhong** [comment]
> LGTM. e2e tests are a flake due to out of CPU quota. Will wait for that to get resolved before I ping the bot again.

I manually cleaned up the leaked VMs (while others are root causing it), so it should be ok to rerun the test now.

**Comment by yujuhong** [comment]
@k8s-bot node e2e test this. issue: #IGNORE

**Comment by dims** [comment]
LGTM 👍

File: pkg/util/procfs/procfs_linux.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
+// +build cgo,linux
```
Comment (luxas) [inline pkg/util/procfs/procfs_linux.go]: Why cgo? I don't think it's required here

Comment (yujuhong) [inline pkg/util/procfs/procfs_linux.go]: don't think we need it. I copied it from other files and forgot to check again.

**Comment by yujuhong** [comment]
Addressed the comment. Re-apply the lgtm.

**Comment by luxas** [comment]
LGTM for me also

**Comment by yujuhong** [comment]
Bumping the priority to fix the release builds.

**Pull request merged**